src/dbos-executor.ts

```diff
@@ -1524,7 +1542,7 @@ export class DBOSExecutor implements DBOSExecutorContext {
 
     const procClassName = this.getProcedureClassName(proc);
     const plainProcName = `${procClassName}_${proc.name}_p`;
-    const procName = this.config.appVersion ? `v${this.config.appVersion}_${plainProcName}` : plainProcName;
+    const procName = globalAppVersion.wasComputed ? plainProcName : `v${globalAppVersion.version}_${plainProcName}`;
```

Comment:
But I assume the stored proc compiler can be updated to use some similar logic to compute the version. That can be a separate PR. @devhawk